### PR TITLE
Use common approach for ensuring serialdev permissions

### PR DESCRIPTION
**Pull request by okurz**
Verification run: http://lord.arch/tests/190

Related progress issue: https://progress.opensuse.org/issues/29940

### CODE REVIEW

File: lib/utils.pm [hunk 1]
```diff
@@ -1304,6 +946,14 @@ sub arrays_differ {
     return 0;
 }
 
+sub ensure_serialdev_permissions {
+    my ($self) = @_;
+    # grant user permission to access serial port until next reboot
+    # and get permanent user permission to access serial port even if reboot
+    assert_script_run "chown $testapi::username /dev/$testapi::serialdev";
+    assert_script_run "gpasswd -a $testapi::username \$(ls -l /dev/$testapi::serialdev | awk \"{print \\\$4}\")";
```
Comment (nicksinger) [inline lib/utils.pm]: IMHO it is better to add the user to a fixed group (e.g. dialout) to catch if permissions ever change.

File: lib/bootloader_setup.pm [hunk 2]
```diff
@@ -519,6 +521,33 @@ sub select_bootmenu_language {
...
+    else {                                           # else go to that kernel thru grub2 advanced options
+        send_key_until_needlematch 'grub2-advanced-options', 'down';
+        send_key 'ret';
+        send_key_until_needlematch "grub2-$kernel-selected", 'down';
+        send_key 'ret';
+    }
+    if (get_var('NOAUTOLOGIN')) {
+        my $ret = assert_screen 'displaymanager', 200;
+        mouse_hide();
+        if (get_var('DM_NEEDS_USERNAME')) {
```
Comment (asmorodskyi) [inline lib/bootloader_setup.pm]: is possible that NOAUTOLOGIN is defined and DM_NEEDS_USERNAME is not ? 
For me it's looks over complicated we can reach same result using single variable.
Less code less bugs :)

Comment (nicksinger) [reply lib/bootloader_setup.pm]: Well… This is code that @okurz just shifted from one file into another so it is not him to blame ;) Technically speaking something that has to do with the "displaymanager" does not belong in a function called "select_kernel" anyway.

File: tests/console/zypper_lr_validate.pm [hunk 3]
```diff
@@ -15,6 +15,326 @@ use strict;
...
+    # Repo is checked for enabled/disabled state. If the information about the
+    # expected state is not delivered to validatelr(), we use some heuristics to
+    # determine the expected state: If the installation medium is a physical
+    # medium and the system is registered to SCC the repo should be disabled
+    # if the system is SLE 12 SP2 and later; enabled otherwise, see PR#11460 and
+    # FATE#320494.
+    my $scc_install_sle12sp2 = check_var('SCC_REGISTER', 'installation') and sle_version_at_least('12-SP2');
+    my $enabled_repo;
+    if ($args->{enabled_repo}) {
+        $enabled_repo = $args->{enabled_repo};
```
Comment (asmorodskyi) [inline tests/console/zypper_lr_validate.pm]: sounds like 
if (true){
b=true ; 
} 
:D

File: tests/console/zypper_lr_validate.pm [hunk 4]
```diff
@@ -15,6 +15,326 @@ use strict;
...
+    }
+    # bsc#1012258, bsc#793709: USB repo is disabled as the USB stick will be
+    # very likely removed from the system.
+    elsif ($args->{uri} =~ m{(cd|dvd|hd):///.*usb-}) {
+        $enabled_repo = 'No';
```
Comment (asmorodskyi) [inline tests/console/zypper_lr_validate.pm]: I understand that you just copy/paste this .... but still it's nightmare. Text representation of boolean is beyond my understanding

File: tests/console/zypper_lr_validate.pm [hunk 5]
```diff
@@ -15,6 +15,326 @@ use strict;
...
+        }
+        else {
+            # SLES12 does not have 'SLES12-Source-Pool' SCC channel
+            unless (($version eq "12") and ($product_channel eq "Source-Pool")) {
+                $cmd
+                  = "zypper lr --uri | awk -F \'|\' -v OFS=\' \' \'{ print \$2,\$3,\$4,\$NF }\' | tr -s \' \' | grep --color \"$product$version\[\[:alnum:\]\[:punct:\]\]*-*$product_channel $product$version\[\[:alnum:\]\[:punct:\]\[:space:\]\]*-*$product_channel $enabled_repo $uri\"";
+            }
+        }
+    }
+    run_scripted_command_slow($cmd, slow_type => 2) if defined $cmd;
```
Comment (asmorodskyi) [inline tests/console/zypper_lr_validate.pm]: In all architectures we can't type this with normal speed ? 
Also if cmd is not defined , we just simply quit ? No error or at least warn message to a log ?
Suggesting to add option which will allow to use normal typing

File: lib/utils.pm (hunk 1 above)
Comment (okurz) [reply lib/utils.pm]: agreed

File: lib/bootloader_setup.pm (hunk 2 above)
Comment (okurz) [reply lib/bootloader_setup.pm]: I moved that code to tests/rt/kmp_modules.pm, the only place where it's used. See the (old) code in action e.g. in https://openqa.suse.de/tests/1305377

**Comment by okurz** [comment]
@asmorodskyi I don't intend the change the behaviour of the test modules I did not change except for moving code around. Your ideas are all valid but at the wrong place. I will try to fix that in another PR

**Comment by asmorodskyi** [comment]
ok , calmn down travis but generally LGTM

File: tests/console/zypper_lr_validate.pm (hunk 5 above)
Comment (okurz) [reply tests/console/zypper_lr_validate.pm]: IIRC we have a ticket to get rid of "run_scripted_command_slow". I think it was not a good idea after all and should be removed -> handled in the ticket

**Pull request merged**